tests/sentinel/tests/06-ckquorum.tcl

```diff
@@ -24,9 +38,15 @@ test "CKQUORUM detects failover authorization cannot be reached" {
         kill_instance sentinel [expr {$i + 1}]
     }
 
-    after 5000
-    catch {[S 0 SENTINEL CKQUORUM mymaster]} err
-    assert_match "*NOQUORUM*" $err
+    # Wait for the current sentinel to monitor that other sentinels are
+    # `down`, so that we can check the quorum directly later.
+    wait_for_condition 1000 50 {
```

Comment:
If system reaches desired state only after  ~40sec, IMO we would like this test to fail. 
Maybe consider no more than 15-20sec of waiting.

Reply:
good point, i also have the same question that the time is too long.
but another thought, since the test only want to verify `SENTINEL CKQUORM` result
and i have only seen this test case fail once, i suppose the old `after 5000` mostly work.

but anyway, you point it out, it make sence to me, will change it to `wait_for_condition 300 50 `

Reply:
i re-simplified the logic, how does it look

Reply:
looks good!